### PR TITLE
Add S3 sink documentation

### LINKED ISSUE #4157: [DOC] Documentation for Data Prepper S3 sink
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

Data Prepper 2.3 adds a new S3 sink. Include this documentation in the Data Prepper -> Pipelines -> Sources section.

This is currently documented in the `README.md`:

https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/s3-sink

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/s3-sink

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Fixes #4157 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        buffer_type: in_memory
+```
+
+## AWS configuration options
+
+Use the following options when customizing the AWS section of the `s3` sink.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`bucket` | Yes | String | The object from which to retrieve and store data. The `name` must match the name of your object store.
```
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:46]: This is in the top-level configuration. That is, it is not under `aws:`.

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+`region` | No | String | The AWS region to use when connecting to S3. Defaults to the [standard SDK behavior to determine the region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
+`sts_role_arn` | No | String | The [AWS Security Token Service](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) (AWS STS) role that the S3 sink assumes when sending request to S3. Default to the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html). 
+`sts_external_id` | No | String | The external ID to attach to AssumeRole requests from AWS STS.
+`max_retries` | No | Integer | The maximum number of times a single request should retry when ingesting data to S3. Defaults to `5`.
```
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:50]: This is in the top-level configuration. That is, it is not under `aws:`.

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+`object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. Defaults to the S3 object `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket.
```
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:51]: This is in the top-level configuration. That is, it is not under `aws:`.

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Threshold configuration options
+
+Use the following options to set ingestion thresholds for the `s3` sink.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`event_count` | Yes | Integer | The maximum number of events the S3 bucket can ingest. 
+`maximum_size` | No | String | The maximum count or size of bytes that the S3 bucket can ingest. Defaults to `50mb`.
```
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:60]: This value is required.

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+`event_collect_timeout` | Yes | String | Sets how long events are collected before ingestion. All values are strings the represent duration, either an ISO_8601 notation string such as `PT20.345S`, or a simple notation such `60s` or `1500ms`.
+
+## buffer_type
```
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:63]: Our general pattern in the documentation is to only create a sub-section in the documentation when there is a sub-section in the YAML.

This is going against that pattern. So this makes it seem like it follows underneath.

e.g.

```
buffer_type:
  local_file: ???
```

I recommend that you move this into a top-level configuration section.

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            path_prefix: my-elb/%{yyyy}/%{MM}/%{dd}/
+        threshold:
+          event_count: 2000
+          maximum_size: 50mb
+          event_collect_timeout: 15s
+        codec:
+          ndjson:
+        buffer_type: in_memory
+```
+
```
Comment (dlvenable) [inline _data-prepper/pipelines/configuration/sinks/s3.md:39]: We are missing a top-level configuration section. We should add that, and it should indicate what options are available. I have a few other comments about other items that should get moved up there.

See the S3 source for the current pattern to use.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+## AWS configuration options
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/sinks/s3.md:40]:
```suggestion
## Configuration 
```

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+Use the following options when customizing the AWS section of the `s3` sink.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/sinks/s3.md:42]:
```suggestion
Use the following options when customizing the section of the `s3` sink.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md (hunk 4 above)
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/sinks/s3.md:60]:
```suggestion
`maximum_size` | Yes | String | The maximum count or size of bytes that the S3 bucket can ingest. Defaults to `50mb`.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        buffer_type: in_memory
+```
+
+## Configuration 
+
+Use the following options when customizing the section of the `s3` sink.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`bucket` | Yes | String | The object from which to retrieve and store data. The `name` must match the name of your object store.
```
Comment (cwillum) [inline _data-prepper/pipelines/configuration/sinks/s3.md:46]:
```suggestion
`bucket` | Yes | String | The object from which the data is retrieved and then stored. The `name` must match the name of your object store.
```

File: _data-prepper/pipelines/configuration/sinks/s3.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+`region` | No | String | The AWS region to use when connecting to S3. Defaults to the [standard SDK behavior to determine the region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
+`sts_role_arn` | No | String | The [AWS Security Token Service](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) (AWS STS) role that the S3 sink assumes when sending request to S3. Default to the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
```
Comment (cwillum) [inline _data-prepper/pipelines/configuration/sinks/s3.md:48]:
```suggestion
`sts_role_arn` | No | String | The [AWS Security Token Service](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) (AWS STS) role that the S3 sink assumes when sending a request to S3. Defaults to the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html). 
```

**Review by cwillum — APPROVED** [review]
LGTM

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

**Comment by dlvenable** [comment]
Thanks @Naarcha-AWS for making the plugin documentation more consistent!

**Pull request merged**